src/formats/MMTF.cpp

```diff
@@ -0,0 +1,118 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+
+#include "chemfiles/formats/MMTF.hpp"
+
+#include "chemfiles/ErrorFmt.hpp"
+#include "chemfiles/File.hpp"
+#include "chemfiles/Frame.hpp"
+#include "chemfiles/utils.hpp"
+#include "chemfiles/warnings.hpp"
+
+using namespace chemfiles;
+
+template<> FormatInfo chemfiles::format_information<MMTFFormat>() {
+    return FormatInfo("MMTF").with_extension(".mmtf").description(
+        "MMTF (RCSB Protein Data Bank) binary format"
+    );
+}
+
+MMTFFormat::MMTFFormat(const std::string& path, File::Mode mode)
+  : file_(path, mode),
+    modelIndex_(0), chainIndex_(0), groupIndex_(0), atomIndex_(0),
+    atomSkip_(0) {
+}
+
+size_t MMTFFormat::nsteps() {
+    return file_->numModels;
+}
+
+void MMTFFormat::read_step(const size_t step, Frame& frame) {
+
+    if (step >= nsteps()) {
```

Comment:
This is already checked by the [Trajectory](https://github.com/chemfiles/chemfiles/blob/deb490b7aaef9c7fddf77bb3ee371b7d71f4d223/src/Trajectory.cpp#L64-L69) class

Reply:
I've removed the check.